File: services/lineage-service/src/domain/executor.rs
use serde_json::Value;
use uuid::Uuid;

use crate::{AppState, models::pipeline::Pipeline, models::run::PipelineRun};

/// `lineage-service` owns lineage reads and workflow fan-out; it does not
/// embed the Pipeline Builder execution engine. Pipeline build candidates
/// therefore surface as skipped until this service grows a dedicated
/// execution handoff.
pub async fn start_pipeline_run(
    _state: &AppState,
    pipeline: &Pipeline,
    _started_by: Option<Uuid>,
    _trigger_type: &str,
    _from_node_id: Option<String>,
    _retry_of_run_id: Option<Uuid>,
    _attempt_number: i32,
    _distributed_worker_count: usize,
    _skip_unchanged: bool,
    _context: Value,
) -> Result<PipelineRun, String> {
    Err(format!(
        "pipeline lineage build handoff is not wired in lineage-service for pipeline {}",
        pipeline.id
    ))
}

File: services/lineage-service/internal/lineage/executor.go
package lineage

import (
	"context"
	"encoding/json"
	"fmt"

	"github.com/google/uuid"

	"github.com/openfoundry/openfoundry-go/services/lineage-service/internal/models"
)

// StartPipelineRun ports `domain::executor::start_pipeline_run`.
//
// The Rust source is also a stub:
//
//	"pipeline lineage build handoff is not wired in lineage-service for pipeline {id}"
//
// `lineage-service` owns lineage reads + workflow fan-out; the
// pipeline build engine lives in pipeline-build-service. The lineage
// build trigger therefore reports the pipeline as `failed` with the
// same error string until a handoff RPC lands.
func StartPipelineRun(_ context.Context, _ *AppState, pipeline *models.Pipeline, _ *uuid.UUID, _, _ string, _ *string, _ *uuid.UUID, _ int32, _ int, _ bool, _ json.RawMessage) (*models.PipelineRun, error) {
	return nil, fmt.Errorf("pipeline lineage build handoff is not wired in lineage-service for pipeline %s", pipeline.ID)
}
